compiler-plugin/src/test/java/io/ballerina/stdlib/persist/compiler/CompilerPluginTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.stdlib.persist.compiler;
+
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.RecordTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * Tests persist compiler plugin.
+ */
+public class CompilerPluginTest {
+
+    @Test
+    public void testIncrementValue() {
```

Comment:
Why did you waste time to create this test cases, which includes you overwriting SyntaxCodeAnalysisContext.java?

What I suggested was to simply call `Utils.getStringLiteralToken` method in the tests. https://github.com/ballerina-platform/module-ballerina-persist/blob/main/compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/Utils.java#L62-L65

This will just be a dummy test for our purposes, which is to make jacaco recognize Constants.java as a class consisting of only static members

Reply:
First, I checked with this method, but only this method kit by the jacoco, and all other class coverage is 0. I couldn't fix that issue also.
So, I tried this as it will use many classes. But this is also the same behavior.

Reply:
From the first try,  seems like the issue is not because the test cases were in different module. 

Shouldn't we look into why jacoco is generating report for Constants class with 0 coverage, rather than writing test after test, when we know it is going to fail?

1. Look into how many lines are recognised for coverage.
2. Why those are being recognised while others aren't.
3. Can we write test cases to cover them? Is it a jacoco limitation or something from our side?
4. If not IMO its better to leave w/o any tests rather than write tests.

Reply:
JaCoCo measures test coverage based on the percentage of bytecode. But declaring a static final primitive or string constant does not generate any bytecode to execute. So, we get 0% coverage.

Reference: https://stackoverflow.com/questions/30121265/how-to-add-static-member-variables-jacoco-test-coverage 

So, removed the unit test in 349ddd4 as no use to add a test.

Reply:
Ack